keras_core/layers/layer.py

```diff
@@ -1353,7 +1353,7 @@ def get_shapes_dict(call_spec):
     """
     shapes_dict = {}
     for k, v in call_spec.tensor_arguments_dict.items():
-        if k == "mask" or k.startswith("mask_"):
+        if k == "mask" or k.endswith("_mask"):
```

Comment:
Alternately, since we are now selectively matching call and build/compute_output_shape argument, we could just remove this entirely.

This would leave mask arguments in the shapes dict, and if people did want to write `build(inputs_shape, mask_shape)` they could. If not the `mask_shape` will just be dropped. @fchollet let me know if we would rather go that way. Seems simpler.

Reply:
IMO no need to include masks at all. Your fix looks good.